src/pytorch_lightning/core/module.py

```diff
@@ -604,7 +613,7 @@ def forward(self, *args, **kwargs) -> Any:
         """
         return super().forward(*args, **kwargs)
 
-    def training_step(self, *args, **kwargs) -> STEP_OUTPUT:
+    def training_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:
```

Comment:
I agree with above and would add that a follow-on type hints project be added to create additional bespoke Lightning types to help uniformity - as is reasonably possible - across files for explicitly named args and kwargs.

Also noting that these types are pushed to docs, under the parameters block; which raises the question of - should Dict, List, Mappings be left simple - as Dict and not Dict[str, List[...]] or some other variation - for the sake of brevity in docs.

Reply:
I would go with `Any` for now. It is weird to leave this unannotated when every other method in `ModelHooks` has annotations that might also be undesired. To me, it's an all-or-nothing decision.

I wonder if things could be improved by using `typing.overloads` so that we don't have to just allow `*args, **kwargs`. But the simplest solution right now is to add `Any` now

Reply:
I vote for keeping it unannotated.